### ISSUE TITLE
Tooltips should fade away on keyboard event

### ISSUE THREAD

**Issue by yarozar** [atom/atom#17431]
### Prerequisites

    * [X] Reproduced the problem in Safe Mode: https://flight-manual.atom.io/hacking-atom/sections/debugging/#using-safe-mode
    * [X] Followed all applicable steps in the debugging guide: https://flight-manual.atom.io/hacking-atom/sections/debugging/
    * [X] Checked the FAQs on the message board for common solutions: https://discuss.atom.io/c/faq
    * [X] Checked that your issue isn't already filed: https://github.com/issues?utf8=✓&q=is%3Aissue+user%3Aatom
    * [X] Checked that there is not already an Atom package that provides the described functionality: https://atom.io/packages

### Description

Tooltips like for example file path when hovering over tab name at the top or tooltip "Click to fetch" when over "Fetch" at the bottom should GO AWAY the moment keyboard event happens (typing anything or simple arrow keys clicks, etc) even if mouse cursor is still over the "tooltip triggering area".

Otherwise, tooltip becomes annoyance instead of helping hand, obstructing my workspace.

### Steps to Reproduce

---
*Edit by @rsese to add alternate repro steps*

Using atom/atom as an example:

1. `atom --safe .`
2. Open `SUPPORT.md`
3. Hover over the tab for the file so the path tooltip appears
4. Start typing

![tooltip-hiding](https://user-images.githubusercontent.com/734194/40796751-b515ac46-64cb-11e8-84da-59ae97dbf614.gif)

---

1. Type in workspace document.
2. Hover mouse over "tooltip triggering area", wait for tooltip to appear.
3. Continue typing while not moving away the mouse.

**Expected behavior:** Tooltip(s) should be gone either on mouse hover away OR when any input event happens like keyboard click, mouse click (?), etc.

**Actual behavior:** Tooltip keeps in the place until I move mouse away only.

**Reproduces how often:** 100%

### Versions

macOS High Sierra 10.13.4

```
$ atom --version
Atom    : 1.27.1
Electron: 1.7.15
Chrome  : 58.0.3029.110
Node    : 7.9.0

$ apm --version
apm  1.19.0
npm  3.10.10
node 6.9.5 x64
atom 1.27.1
python 2.7.10
git 2.15.1
```

### Additional Information

<img width="530" alt="atom_tooltip_bug" src="https://user-images.githubusercontent.com/2981562/40766917-b2da9cf0-64b9-11e8-9b74-ca57526c636d.png">

**Comment by rsese** [comment]
Thanks for reaching out!  I see the behavior you're describing but can you update the issue body and fill out the issue template? The information is super helpful for us when triaging issues. Once you add those details we can re-open.

---

We require the template to be filled out on all new issues and pull requests so if you can open a new issue with the template filled out or update the issue body with the issue template info, that would be helpful for us. We do this so that we can be certain we have all the information we need to address your submission efficiently. This allows the maintainers to spend more time fixing bugs, implementing enhancements, and reviewing and merging pull requests.

Thanks for understanding and meeting us half way :grinning:

**Comment by yarozar** [comment]
@rsese- I have updated bug report to the template, please reopen. Thanks!

**Comment by rsese** [comment]
Thanks @yarozar :bow:  I went ahead and added alternate steps to reproduce since specific and exact steps are preferred (as well as steps that don't require community packages).  I also added a GIF to show the behavior.

**Comment by cacheflow** [comment]
I'll take a look at this.

**Comment by cacheflow** [comment]
PR open if anyone wants to take a look: https://github.com/atom/atom/pull/17651

**Comment by yarozar** [comment]
@rsese I still observe the issue after 1.30 version release ...

**Comment by cacheflow** [comment]
Pushing a fix now. @yarozar.

**Comment by rsese** [comment]
Thanks @yarozar, I can reproduce too with 1.32.0-nightly0 on macOS 10.12.6, I do believe the fix is in 1.30.

I'll mention this to the team but if you have any ideas @cacheflow please let me know :v:

**Comment by cacheflow** [comment]
@rsese, I think it's because event propagation was not occurring or getting stopped before it could bubble up to the window, so I added a `{capture: true}`, which seems to fix the problem. However, I'm curious as to why hiding tooltips on resizing worked without capture, but not for ``keyDown` events?
